### PR TITLE
feat(VDataTable): add contexmenu:row event for data table

### LINKED ISSUE #9382: [Feature Request] Add @contextmenu:row event for v-data-table
### Problem to solve
I want to implement a custom context menu for my data table triggered with a right click, and to do so, I must currently override the `item` slot and add a `@contextmenu` event listener to the `<tr>` element.

However, this means I need to manually define the content for every column (e.g `<td v-text="item.prop"></td>`), instead of allowing the data table to handle the columns with its default rendering. This can become problematic and difficult to maintain for tables with many columns, and you lose the benefit of having the data table sort based on the actual data values rather than the displayed string (e.g if I have a formatted date string or currency amount), which forces you to write custom sort functions.

### Proposed solution
Proposed solution is to add a `contextmenu:row` event that functions like the `click:row` event to `v-data-table`, so that the right click event is easily captured without having to specify every column in the item slot.

### PR DESCRIPTION
**Pull request by Leespiker**
## Description

add contextmenu:row event for data table

## Motivation and Context

resolves #9382 

## How Has This Been Tested?

unit & visually

## Markup:

<details>

```vue
<template>
  <v-container>
    <v-data-table
      class="elevation-1"
      :headers="headers"
      :items="desserts"
      item-key="name"
      @contextmenu:row="onRowContextmenu"
    />

    <v-menu
      v-model="showMenu"
      :position-x="menuX"
      :position-y="menuY"
      absolute
      offset-y
    >
      <v-list>
        <v-list-item
          v-for="item in items"
          :key="item.title"
          @click=""
        >
          <v-list-item-title>{{ item.title }}</v-list-item-title>
        </v-list-item>
      </v-list>
    </v-menu>
  </v-container>
</template>

<script>
  export default {
    data: () => ({
      desserts: [
        {
          name: 'Frozen Yogurt',
          calories: 159,
          fat: 6.0,
          carbs: 24,
          protein: 4.0,
          iron: '1%',
        },
        {
          name: 'Ice cream sandwich',
          calories: 237,
          fat: 9.0,
          carbs: 37,
          protein: 4.3,
          iron: '1%',
        },
        {
          name: 'Eclair',
          calories: 262,
          fat: 16.0,
          carbs: 23,
          protein: 6.0,
          iron: '7%',
        },
        {
          name: 'Cupcake',
          calories: 305,
          fat: 3.7,
          carbs: 67,
          protein: 4.3,
          iron: '8%',
        },
        {
          name: 'Gingerbread',
          calories: 356,
          fat: 16.0,
          carbs: 49,
          protein: 3.9,
          iron: '16%',
        },
        {
          name: 'Jelly bean',
          calories: 375,
          fat: 0.0,
          carbs: 94,
          protein: 0.0,
          iron: '0%',
        },
        {
          name: 'Lollipop',
          calories: 392,
          fat: 0.2,
          carbs: 98,
          protein: 0,
          iron: '2%',
        },
        {
          name: 'Honeycomb',
          calories: 408,
          fat: 3.2,
          carbs: 87,
          protein: 6.5,
          iron: '45%',
        },
        {
          name: 'Donut',
          calories: 452,
          fat: 25.0,
          carbs: 51,
          protein: 4.9,
          iron: '22%',
        },
        {
          name: 'KitKat',
          calories: 518,
          fat: 26.0,
          carbs: 65,
          protein: 7,
          iron: '6%',
        },
      ],
      headers: [
        {
          align: 'start',
          sortable: false,
          text: 'Dessert (100g serving)',
          value: 'name',
        },
        {
          text: 'Calories',
          value: 'calories'
        },
        {
          text: 'Fat (g)',
          value: 'fat'
        },
        {
          text: 'Carbs (g)',
          value: 'carbs'
        },
        {
          text: 'Protein (g)',
          value: 'protein'
        },
        {
          text: 'Iron (%)',
          value: 'iron'
        },
      ],
      items: [],
      menuX: 0,
      menuY: 0,
      showMenu: false,
    }),
    methods: {
      onRowContextmenu (event, data) {
        event.preventDefault()
        this.items = Object.entries(data).map(([key, value]) => {
          return {
            title: `${key}: ${typeof value === 'function' ? value.toString() : JSON.stringify(value)}`
          }
        }).sort((item1, item2) => item1.title.localeCompare(item2.title))
        this.showMenu = false
        this.menuX = event.clientX
        this.menuY = event.clientY
        this.$nextTick(() => {
          this.showMenu = true
        })
      }
    },
  }
</script>
```
</details>

## Docs
<details>

unknown

 </details>

## Types of changes

- [X] New feature (non-breaking change which adds functionality)

## Checklist:

- [X] The PR title is no longer than 64 characters.
- [X] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [X] My code follows the code style of this project.
- [X] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/api-generator/src/maps/v-data-table.js [hunk 1]
```diff
@@ -17,7 +17,16 @@ const TableHeader = {
 }
 
 const DataTableEvents = [
-  { name: 'click:row', source: 'v-data-table', value: 'any, { select: (value: boolean) => void, isSelected: boolean, expand: (value: boolean) => void, isExpanded: boolean }' },
+  {
+    name: 'click:row',
+    source: 'v-data-table',
```
Comment (johnleider) [inline packages/api-generator/src/maps/v-data-table.js:22]: I know some of the @vuetifyjs/core-team enjoys very long lines but please format this and other instances 🙏

File: packages/vuetify/src/components/VDataTable/__tests__/VDataTable.spec.ts [hunk 2]
```diff
@@ -348,6 +348,24 @@ describe('VDataTable.ts', () => {
     expect(fn).toHaveBeenCalled()
   })
 
+  it('should emit event when contextmenu is triggered on internally created row', async () => {
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/components/VDataTable/__tests__/VDataTable.spec.ts:351]: https://github.com/vuetifyjs/vuetify/pull/11039/files#r403810115

**Comment by KaelWD** [comment]
Should we just check `this.$listeners` for anything ending in `:row` and pass that through instead of a few explicit events?

**Comment by Leespiker** [comment]
@KaelWD It's definitely possible. However, I'm not sure how that would look in the docs. Also, emitting events only when they're listened for could potentially be a bad practice. Would be interested to hear others' thoughts. @vuetifyjs/contributors

**Comment by nekosaur** [comment]
I think I'm in agreement with @Leespiker that it's something we can look at for v3

**Pull request merged**